saleor/graphql/checkout/mutations.py

```diff
@@ -457,19 +456,19 @@ def clean_input(
         )
 
         if variants and quantities:
-            for variant, quantity in zip(variants, quantities):
-                try:
-                    checkout = add_variant_to_checkout(
-                        checkout_info, variant, quantity, replace=replace
-                    )
-                except InsufficientStock as exc:
-                    error = prepare_insufficient_stock_checkout_validation_error(exc)
-                    raise ValidationError({"lines": error})
-                except ProductNotPublished as exc:
-                    raise ValidationError(
-                        "Can't add unpublished product.",
-                        code=exc.code,
-                    )
+            try:
+                checkout = add_variants_to_checkout(
+                    checkout,
+                    variants,
+                    quantities,
+                    checkout_info.channel.slug,
```

Comment:
In line 448 we also fetch channel slug. Maybe we could store it in a local variable and fetch it one time in this function?

Reply:
On local i see this is cached already, it doesn't query the DB second time. However yes, it seems better to simply assign it if it is an ORM value used multiple times to avoid performance issues when maintaining it.